packages/nitrogen/src/syntax/createType.ts

```diff
@@ -257,10 +257,6 @@ export function createType(type: TSMorphType, isOptional: boolean): Type {
       throw new Error(
         `Anonymous objects cannot be represented in C++! Extract "${type.getText()}" to a separate interface/type declaration.`
       )
-    } else if (type.isStringLiteral()) {
```

Comment:
@mrousavy I feel like this was here for a reason.  While these changes "work" and create generated files without error, I sense they are breaking another part of nitro handling strings/enums/etc.